src/main/java/org/broadinstitute/dsm/model/BSPKit.java

```diff
@@ -0,0 +1,150 @@
+                    if (participant != null) {
+                        boolean specialBehavior = InstanceSettings.shouldKitBehaveDifferently(participant, received);
+                        if (specialBehavior) {
+                            //don't trigger ddp to sent out email, only email to study staff
+                            if (InstanceSettings.TYPE_NOTIFICATION.equals(received.getType())) {
+                                String message = "Kit of participant " + bspKitQueryResult.getBspParticipantId() + " was received by GP. <br> " +
+                                        "CollaboratorSampleId:  " + bspKitQueryResult.getBspSampleId() + " <br> " +
+                                        received.getValue();
+                                notificationUtil.sentNotification(bspKitQueryResult.getNotificationRecipient(), message, NotificationUtil.UNIVERSAL_NOTIFICATION_TEMPLATE, NotificationUtil.DSM_SUBJECT);
+                            }
+                            else {
+                                logger.error("Instance settings behavior for kit was not known " + received.getType());
+                            }
+                        }
+                        else {
+                            triggerDDP(conn, bspKitQueryResult, firstTimeReceived, kitLabel);
+                        }
+                    }
+                }
+                else {
+                    triggerDDP(conn, bspKitQueryResult, firstTimeReceived, kitLabel);
+                }
+
+                String bspParticipantId = bspKitQueryResult.getBspParticipantId();
+                String bspSampleId = bspKitQueryResult.getBspSampleId();
+                String bspMaterialType = bspKitQueryResult.getBspMaterialType();
+                String bspReceptacleType = bspKitQueryResult.getBspReceptacleType();
+                int bspOrganism;
+                try {
+                    bspOrganism = Integer.parseInt(bspKitQueryResult.getBspOrganism());
+                }
+                catch (NumberFormatException e) {
+                    throw new RuntimeException("Organism " + bspKitQueryResult.getBspOrganism() + " can't be parsed to integer", e);
+                }
+
+                logger.info("Returning info for kit w/ label " + kitLabel + " for " + bspKitQueryResult.getInstanceName());
+                return new BSPKitInfo(bspKitQueryResult.getBspCollection(),
+                        bspOrganism,
+                        "U",
+                        bspParticipantId,
+                        bspSampleId,
+                        bspMaterialType,
+                        bspReceptacleType);
+            }
+            else {
+                throw new RuntimeException("No participant id for " + kitLabel + " from " + bspKitQueryResult.getInstanceName());
+            }
+        });
+    }
+
+    private static void writeSampleReceivedToES(DDPInstance ddpInstance, BSPKitQueryResult bspKitInfo) {
```

Comment:
access modifier static can be removed easily, since BSPKit is following OOP and this method is part of its internal business logic.